main/version.h

```diff
@@ -3,9 +3,9 @@
 
 #include <stdint.h>
 
-const char *version = "0.9.0";
-const uint16_t version_major = 0;
-const uint16_t version_minor = 9;
-const uint16_t version_patch = 0;
+#define VERSION "0.9.0"
```

Comment:
One advantage of using actual memory is that you can see (and check/extract) the version from the binary without running any code.

Reply:
One disadvantage is that it's hard to share between files. The compiler didn't like sharing the integers. I suppose a getter would work. What do you think?